.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,286 @@
+              exit 1
+          fi
+
+  ui-integration-tests:
+    needs: build-engine-docker-image
+    runs-on: ubuntu-latest
+    name: "UI Integration Tests - Grafana: ${{ matrix.grafana-image-tag }}"
+    strategy:
+      matrix:
+        grafana-image-tag:
+          - 9.2.6
+          - main
+          - latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Download engine Docker image
+        uses: actions/download-artifact@v2
+        with:
+          name: oncall-engine
+          path: /tmp
+
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1.3.0
+        with:
+          config: ./helm/kind.yml
+
+      - name: Load image on the nodes of the cluster
+        run: kind load image-archive --name=chart-testing /tmp/oncall-engine.tar
+
+      # yarn caching doesn't properly work with subdirectories hence the following two steps
+      # which calculate a cache key and restore the cache manually
+      # see this GH issue for more details https://github.com/actions/setup-node/issues/488#issue-1231822552
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14.17.0
+          cache: "yarn"
+          cache-dependency-path: grafana-plugin/yarn.lock
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "dir=$(yarn config get cacheFolder)" >> $GITHUB_OUTPUT
+        shell: bash
+        working-directory: ./grafana-plugin
+
+      - name: Restore yarn cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: yarn-cache-folder-${{ hashFiles('**/yarn.lock', '.yarnrc.yml') }}
+          restore-keys: |
+            yarn-cache-folder-
+
+      # https://stackoverflow.com/a/62244232
+      # --prefer-offline tells yarn to use cached downloads (in the cache directory mentioned above)
+      # during installation whenever possible instead of downloading from the server.
+      - name: Install dependencies
+        working-directory: ./grafana-plugin
+        run: yarn install --frozen-lockfile --prefer-offline
+
+      # build the plugin frontend
+      - name: Build plugin frontend
+        working-directory: ./grafana-plugin
+        run: yarn build:dev
+
+      # TODO: need to figure out how to properly mount the grafana-plugin/dist
+      # to /var/lib/grafana/plugins/grafana-plugin inside of the grafana container
+      # right now the grafana container is actually pulling the latest published OnCall plugin from GCS and not
+      # using the local build (which means we're not testing any changes that would be made in the PR being tested)
+      # need to somehow tweak the grafana.extraVolumeMounts value to work properly
+      # https://github.com/grafana/helm-charts/blob/main/charts/grafana/README.md#example-of-extravolumemounts
```

Comment:
this is the major outstanding TODO. I've been struggling to get this to work locally, open to suggestions